planner/core/logical_plan_builder.go

```diff
@@ -2668,19 +2670,22 @@ func (b *PlanBuilder) buildProjectionForWindow(p LogicalPlan, expr *ast.WindowFu
 
 	proj.SetSchema(schema)
 	proj.SetChildren(p)
-	return proj, propertyItems, newArgList, nil
+	return proj, propertyItems[:lenPartition], propertyItems[lenPartition:], newArgList, nil
 }
 
 func (b *PlanBuilder) buildWindowFunction(p LogicalPlan, expr *ast.WindowFuncExpr, aggMap map[*ast.AggregateFuncExpr]int) (*LogicalWindow, error) {
-	p, byItems, args, err := b.buildProjectionForWindow(p, expr, aggMap)
+	p, partitionBy, orderBy, args, err := b.buildProjectionForWindow(p, expr, aggMap)
 	if err != nil {
 		return nil, err
 	}
 
 	desc := aggregation.NewWindowFuncDesc(b.ctx, expr.F, args)
+	// TODO: Check if the function is aggregation function after we support more functions.
```

Comment:
What is this TODO mean?

Reply:
In the following line, we warp the cast for agg functions, but it will panic for other window functions.